### DISCUSSION TITLE
schedule doesn't support year - how do I do this?

### DISCUSSION THREAD

**Discussion by nottrobin** [renovatebot/renovate#8226] [Request Help] [answered]
I wanted to make Renovate update yearly. I [tried](https://github.com/canonical-web-and-design/renovate-websites/pull/3/files) "on the first day of the year" after testing it on [a later.js checker](https://codepen.io/antonlydike/full/KgygpA). But Renovate [complained](https://github.com/canonical-web-and-design/renovate-websites/issues/4).

What could I do instead? Would "on the first day of january" work?

**Comment by nottrobin** [comment, marked as answer]
I think maybe [changing to](https://github.com/canonical-web-and-design/renovate-websites/commit/c373189b87c9f1aaf64eb0057e8ade7cbe8e8d5d) "on the first day of january" [worked](https://github.com/canonical-web-and-design/renovate-websites/issues/4).

**Comment by HonkingGoose** [comment]
## Your config now:

What you have now runs on New Year's Day:

```json
    "packageRules": [
        {
            "groupName": "all dependencies",
            "groupSlug": "all",
            "packagePatterns": ["*"],
            "schedule": ["on the first day of january"]
        }
```

## What I think you actually wanted:

```json
    "packageRules": [
        {
            "groupName": "all dependencies",
            "groupSlug": "all",
            "packagePatterns": ["*"],
            "schedule": ["every 12 months on the first day of the month"]
        }
```

This should give you updates spaced a year apart.
I think you would get a update PR from Renovate on the first of February, and afterwards once a year in February (12 months later).

It seems to be supported, according to: https://docs.renovatebot.com/configuration-options/#schedule, see the `every 3 months on the first day of the month` in the .
I'm just changing the number from 3 to 12.

**Reply by nottrobin** [reply]
Can I just say "every 12 months"?

**Reply by HonkingGoose** [reply]
I think "every 12 months" will also work. You would then get PRs once every 12 months, so that they're spread out in time across your repositories.

So the choice is between bundled or unbundled updates:

1. I want all update PRs on the first day of the new year: use my schedule from above.
2. I want update PRs once every 12 months, don't want to bundle them on a date: use "every 12 months".

**Reply by nottrobin** [reply]
Yes that's what I'm thinking. I assumed that "every 12 months on the first day of the month" as you suggested would also space the updates out across the year, depending on the month that they were last updated. Is that wrong? I just thought why necessarily tie them to the first.

**Reply by HonkingGoose** [reply]
>  I assumed that "every 12 months on the first day of the month" as you suggested would also space the updates out across the year, depending on the month that they were last updated.

I think it **does** space out based on the month they were last updated.
The difference is that `on the first day of the month` is an additional constraint on the `every 12 months`.
So you can choose whether or not you care about having those PRs on a random day of the month, or on the first of the month. 😄
